src/views/AnnualReport.vue

```diff
@@ -859,9 +869,19 @@ export default {
           }
           filing = res.data.filing
           this.haveChanges = false
-        }).catch(error => {
+        }).catch(async error => {
           if (error && error.response && error.response.status === PAYMENT_REQUIRED) {
-            this.paymentErrorDialog = true
+            const errCode = this.getErrorCode(error)
+            if (errCode) {
+              const errObj = await this.getErrorObj(errCode.payment_error_type)
+              if (errObj) {
+                this.bcolErrMsg = errObj.detail
+                this.bcolTitle = errObj.title
+                this.bcolErrorDialog = true
```

Comment:
Instead of having 3 different properties for this dialog, have you considered merging them? For instance "if errorObj isn't null then display the dialog". Then you only have to pass a single prop to the dialog. Then, inside the dialog, since you won't be able to use the boolean as a v-model, instead create a boolean getter (true if errorObj isn't null) and use that as a value prop to display the dialog (see https://vuetifyjs.com/en/components/dialogs/). Ask if you want a code example.

Reply:
I actually did consider this, but I didn't see too many components taking advantage of passing an object/dto as a parameter so I figured I should keep them separate. Will update

Reply:
Fixed. Looks way cleaner!